Content.Server/Chemistry/ReagentEffects/MakeSentient.cs

```diff
@@ -16,25 +16,16 @@ public override void Effect(ReagentEffectArgs args)
         var entityManager = args.EntityManager;
         var uid = args.SolutionEntity;
 
-        // This makes it so it doesn't affect things that are already sentient
```

Comment:
Shouldn't you keep this first check, as opposed to the second check, because RemoveComponent gets called?

Reply:
Intentional.

The checks currently just see if the entity has a mind, rather than something more reasonable like if it has a ghost role. If only the first check is called, what happens is the accent removal applies to, say, mice, whose roles haven't been taken, but it _doesn't_ remove the accents of mice whose roles _have_ been taken.

Removing the accents and then checking for mind makes the accent removal always apply, which gives cognizine the secondary use of giving creatures the power of speech (since ghostrole mice are supposedly not intelligent the cognizine can be seen as simply doing its job) 

It's not a nice system but it's better than the inconsistency seen before.

I hope I understood your question.

Reply:
It took me several rereadings of this explanation before I understood this, so perhaps you should take some time and revise this explanation and add it to comments. As you yourself state, "comment why, not what", and this is completely non-obvious.